src/main/java/bio/terra/service/dataset/flight/ingest/IngestParseJsonFileStep.java

```diff
@@ -58,18 +53,24 @@ public StepResult doStep(FlightContext flightContext)
     workingMap.put(IngestMapKeys.TABLE_SCHEMA_FILE_COLUMNS, fileRefColumnNames);
 
     List<String> errors = new ArrayList<>();
-    List<JsonNode> fileLineJson =
-        gcsFileLines.stream()
-            .map(
-                content -> {
-                  try {
-                    return objectMapper.readTree(content);
-                  } catch (JsonProcessingException ex) {
-                    errors.add(ex.getMessage());
-                    return null;
-                  }
-                })
-            .collect(Collectors.toList());
+    Set<BulkLoadFileModel> fileModels =
+        IngestUtils.getJsonNodesStreamFromFile(gcsPdao, objectMapper, ingestRequest, dataset)
+            .flatMap(pair -> IngestUtils.resolveJsonNodeCollectError(pair, errors))
+            .flatMap(
+                node ->
+                    fileRefColumnNames.stream()
+                        .flatMap(
+                            columnName -> {
+                              JsonNode fileRefNode = node.get(columnName);
+                              if (Objects.nonNull(fileRefNode) && fileRefNode.isObject()) {
```

Comment:
I think the intent is to use `Objects.nonNull()` when a function argument is needed. In this case comparing against `null` directly is just as clear and doesn't require a function call.
```suggestion
                              if (fileRefNode != null && fileRefNode.isObject()) {
```